contracts/metagov/utils/VoteEscrowTokenManager.sol

```diff
@@ -0,0 +1,92 @@
+    function increase_unlock_time(uint256 unlock_time) external;
+    function withdraw() external;
+    function locked__end(address) external view returns (uint256);
+    function checkpoint() external;
+}
+
+/// @title Vote-escrowed Token Manager
+/// Used to permanently lock tokens in a vote-escrow contract, and refresh
+/// the lock duration as needed.
+/// @author Fei Protocol
+abstract contract VoteEscrowTokenManager is CoreRef {
+
+    /// @notice The maximum lock duration of veTokens
+    uint256 private immutable MAXTIME;
+    /// @notice One week, in seconds. Vote-locking is rounded down to weeks.
+    uint256 private constant WEEK = 7 * 86400; // 1 week, in seconds
+
+    /// @notice The vote-escrowed token address
+    IVeToken public immutable veToken;
+
+    /// @notice The token address
+    IERC20 public immutable liquidToken;
+
+    /// @notice VoteEscrowTokenManager token Snapshot Delegator PCV Deposit constructor
+    /// @param _liquidToken the token to lock for vote-escrow
+    /// @param _veToken the vote-escrowed token used in governance
+    /// @param _maxTime maximum time tokens can be vote-escrowed for
+    constructor(
+        IERC20 _liquidToken,
+        IVeToken _veToken,
+        uint256 _maxTime
+    ) {
+        liquidToken = _liquidToken;
+        veToken = _veToken;
+        MAXTIME = _maxTime;
+    }
+
+    /// @notice Deposit tokens to get veTokens. Set lock duration to MAXTIME.
+    /// The only way to withdraw tokens will be to pause this contract
+    /// for MAXTIME and then call exitLock().
+    function lock() external whenNotPaused onlyTribeRole(TribeRoles.METAGOVERNANCE_TOKEN_STAKING) {
+        uint256 tokenBalance = liquidToken.balanceOf(address(this));
+        uint256 locked = veToken.locked(address(this));
+        uint256 lockHorizon = ((block.timestamp + MAXTIME) / WEEK) * WEEK;
+
+        // First lock
+        if (tokenBalance != 0 && locked == 0) {
+            liquidToken.approve(address(veToken), tokenBalance);
+            veToken.create_lock(tokenBalance, lockHorizon);
+        }
+        // Increase amount of tokens locked & refresh duration to MAXTIME
+        else if (tokenBalance != 0 && locked != 0) {
+            liquidToken.approve(address(veToken), tokenBalance);
+            veToken.increase_amount(tokenBalance);
+            if (veToken.locked__end(address(this)) != lockHorizon) {
+                veToken.increase_unlock_time(lockHorizon);
+            }
+        }
+        // No additional tokens to lock, just refresh duration to MAXTIME
+        else if (tokenBalance == 0 && locked != 0) {
+            veToken.increase_unlock_time(lockHorizon);
+        }
+        // If tokenBalance == 0 and locked == 0, there is nothing to do.
+    }
+
+    /// @notice Exit the veToken lock. For this function to be called and not
+    /// revert, tokens had to be locked previously, and the contract must have
+    /// been paused for MAXTIME in order to prevent lock extensions
```

Comment:
yes, that's how veTokens are intended to work 😓 also your voting power decreases linearly over time, so if you don't renew your 4-year lock, after 2 years you only have half of your voting power left. Splitting your locks in different tranches is not desirable because you'd need multiple addresses & the tranches that expire soon don't provide voting power anymore